### ISSUE TITLE
Stop reading text for search indexing in 4096 byte blocks

### ISSUE THREAD

**Issue by asmecher** [pkp/pkp-lib#8502]
**Describe the bug**
When the search index tries to index an external file (SearchFileParser) or read from an external process (SearchHelperParser), it reads content to be indexed [in 4096 byte blocks](https://github.com/pkp/pkp-lib/blob/3_3_0-13/classes/search/SearchFileParser.inc.php#L87).

If a UTF8 sequence falls over a 4096 byte boundary, it can lead to an incomplete UTF-8 sequence being detected and the entire string being discarded.

Some possible solutions:
1. Stop specifying a block size in `fgets`; just let it get a line at a time. (May perform more slowly.)
2. Use `stream_get_contents` to read the entire stream at once. (Will require the entire thing being indexed to be loaded in memory at once.)

**What application are you using?**
Any version of OJS, OMP, OPS

**Comment by asmecher** [comment]
Needs performance testing to see whether the potential degradation on the first solution is a) real and b) problematic.

**Comment by jonasraoni** [comment]
@asmecher Just to confirm, the fix didn't address the issue I had, so I'll inspect it deeper 🤔

Anyway, it makes sense to let the `fgets()` to read until it finds a newline, to avoid getting partial sequences, I don't think there will be any noticeable performance issue, especially when indexing through an external tool, but I'll leave a benchmark running to check.

Another alternative that might work is using the `mb_check_encoding()` to read extra characters until the data becomes valid, I'll check if it works.

**Comment by jonasraoni** [comment]
For the sake of completeness, here's the measures from reindexing a big dataset (3300 submissions):
- fgets() without max length
```sh
real    215m22.028s
user    10m45.640s
sys     1m8.370s
```
- stream_get_contents()
```
real    197m10.674s
user    10m3.310s
sys     1m4.630s
```

**Comment by asmecher** [comment]
OK, that's not enough of a difference to worry about -- I'll go ahead and commit the removal of the 4096 byte block size, but @jonasraoni, if you're able to shed more light on what's not indexing in your case, please let me know!

**Comment by jonasraoni** [comment]
Ok! I'll create another issue for it 👌

**Comment by jonasraoni** [comment]
I've just inspected the issue I had more deeply, and the problem I had was due to a limitation of the search implementation.

When searching for "keyword1 keyword2 keywordN" N queries will be issued, each one will bring at most ~500 submissions, and a given submission should be present in all the result sets to be considered as a hit. This way popular keywords in a journal with many submissions are problematic.
